File: desafio-02/vanessamachado93/js/primos.js
function listaPrimos(num) {
	if (num < 2) {
		return false
	}
	for (let i = 2; i < num; i++) {
		if (num % i === 0) {
			return false
		}
	}
	return true
}

let numeros = []
for (let i = 0; i < 10000; i++) {
	numeros[i] = i + 1
}

var resultado = numeros.filter(listaPrimos)

console.log(`Os números primos de 1 a 10000 são:\n ${resultado}.`)

File: desafio-01/simpgda/c/primos.c
#include <stdio.h>

int main() {
    int i = 0;
    int j = 0;
    int primo = 0;
    
    printf("Numeros primos entre 1 e 10000:\n");
    printf("2\n"); // 2 é o único primo par
    
    // Começa do 3 e pula números pares
    for (i = 3; i <= 10000; i += 2) {
        primo = 1;
        for (j = 3; j * j <= i; j += 2) { // só testa divisores ímpares
            if (i % j == 0) {
                primo = 0;
                break;
            }
        }
        if (primo) {
            printf("%d\n", i);
        }
    }
    return 0;
}